src/services/completions.ts

```diff
@@ -167,11 +169,21 @@ namespace ts.Completions {
             return undefined;
         }
         const { name, needsConvertPropertyAccess } = info;
-        Debug.assert(!(needsConvertPropertyAccess && !propertyAccessToConvert));
         if (needsConvertPropertyAccess && !includeInsertTextCompletions) {
             return undefined;
         }
 
+        let insertText: string | undefined;
+        let replacementSpan: TextSpan | undefined;
+        if (kind === CompletionKind.Global && origin && origin.type === "this-type") {
+            insertText = needsConvertPropertyAccess ? `this["${name}"]` : `this.${name}`;

```

Comment:
do you not need replacement span eg. if refering to property x and you want to replace it with this.x ?

Reply:
The default replacement span should already be the identifier itself. We only need to set `replacementSpan` if we are replacing something else, too, such as replacing `.x` with `["x"]`.